src/components/ConnectToQuickbooksOnlineButton/index.native.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {useCallback, useRef, useState} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import type {WebViewNavigation} from 'react-native-webview';
+import {WebView} from 'react-native-webview';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import Button from '@components/Button';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import Modal from '@components/Modal';
+import useLocalize from '@hooks/useLocalize';
+import {getQuickBooksOnlineSetupLink} from '@libs/actions/connections/QuickBooksOnline';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Session} from '@src/types/onyx';
+import type {ConnectToQuickbooksOnlineButtonProps} from './types';
+
+type ConnectToQuickbooksOnlineButtonOnyxProps = {
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+const renderLoading = () => <FullScreenLoadingIndicator />;
+
+function ConnectToQuickbooksOnlineButton({policyID, session}: ConnectToQuickbooksOnlineButtonProps & ConnectToQuickbooksOnlineButtonOnyxProps) {
+    const [isWebViewOpen, setWebViewOpen] = useState<boolean>(false);
+    const [isQuickbooksOnlineReady, setIsQuickbooksOnlineReady] = useState<boolean>(false);
+    const {translate} = useLocalize();
+    const webViewRef = useRef<WebView>(null);
+    const authToken = session?.authToken ?? null;
+
+    const handleNavigationStateChange = useCallback(({url, loading}: WebViewNavigation) => {
+        if (loading || !url.startsWith('https://accounts.intuit.com/app/sign-in')) {
+            return;
+        }
+        setIsQuickbooksOnlineReady(true);
+    }, []);
```

Comment:
For which cases checking the url is needed?

Reply:
We start the flow by opening an expensify URL and after some redirections we get to the `https://accounts.intuit.com` page for sign in, so I was doing this to show a loading spinner while the redirections are happening. Without it, I think the user just see a blank page.

Reply:
I didn't notice a blank page but what you are saying makes sense. Can we put the url in a CONST?

Reply:
Yeah, I tested again special handling and I think it looks the same, maybe it is not necessary.